nano/node/active_transactions.cpp

```diff
@@ -346,73 +350,128 @@ void nano::active_transactions::request_loop ()
-			if (info.block_count > info.confirmation_height && !node.pending_confirmation_height.is_processing_block (info.head))
+			// Prioritize local accounts first
+			std::vector<std::pair<nano::account, nano::account_info>> accounts;
 			{
-				auto num_uncemented = info.block_count - info.confirmation_height;
-				lk.lock ();
-				auto it = priority_cementable_frontiers.find (account);
-				if (it != priority_cementable_frontiers.end ())
-				{
-					if (it->blocks_uncemented != num_uncemented)
-					{
-						// Account already exists and there is now a different uncemented block count so update it in the container
-						priority_cementable_frontiers.modify (it, [num_uncemented](nano::cementable_account & info) {
-							info.blocks_uncemented = num_uncemented;
-						});
-					}
-				}
-				else
+				std::lock_guard<std::mutex> lock (node.wallets.mutex);
+				auto wallet_transaction (node.wallets.tx_begin_read ());
+				for (auto & item : node.wallets.items)
 				{
-					assert (priority_cementable_frontiers_size <= max_priority_cementable_frontiers);
-					if (priority_cementable_frontiers_size == max_priority_cementable_frontiers)
+					auto & wallet (item.second);
+					nano::account_info info;
+					std::lock_guard<std::recursive_mutex> wallet_lock (wallet->store.mutex);
+
+					auto i (wallet->store.begin (wallet_transaction, next_local_frontier_account));
+					auto n (wallet->store.end ());
+					for (; i != n; ++i)
 					{
-						// The maximum amount of frontiers stored has been reached. Check if the current frontier
-						// has more uncemented blocks than the lowest uncemented frontier in the collection if so replace it.
-						auto least_uncemented_frontier_it = priority_cementable_frontiers.get<1> ().end ();
-						--least_uncemented_frontier_it;
-						if (num_uncemented > least_uncemented_frontier_it->blocks_uncemented)
+						auto & account (i->first);
+						if (!node.store.account_get (transaction_a, account, info))
 						{
-							priority_cementable_frontiers.get<1> ().erase (least_uncemented_frontier_it);
-							priority_cementable_frontiers.emplace (account, num_uncemented);
+							// If it exists in normal priority collection delete from there.
+							auto it = priority_cementable_frontiers.find (account);
+							if (it != priority_cementable_frontiers.end ())
+							{
+								priority_cementable_frontiers.erase (it);
+							}
+
+							prioritize_account_for_confirmation (priority_local_cementable_frontiers, priority_local_cementable_frontiers_size, account, info);
+
+							if (timer.since_start () >= time_a)
+							{
+								break;
+							}
 						}
+						next_local_frontier_account = account.number () + 1;
 					}
-					else
+					// Go back to the beginning when we have reached the end of the local accounts and make sure we skip local accounts
+					if (i == n)
 					{
-						priority_cementable_frontiers.emplace (account, num_uncemented);
+						next_local_frontier_account = 0;
```

Comment:
By default should be nano::wallet_store::special_count according to nano::wallet_store::begin () to not iterate over tech fields. Same for default value in header

Reply:
Updated, thanks